src/glados/utils/automated_install.py

```
@@ -0,0 +1,129 @@
+        subprocess.run(["curl", "-L", download_url, "-o", str(installer_path)], check=True)
+        subprocess.run([str(installer_path)], check=True)
+        print("Ollama installed successfully on Windows")
+    except subprocess.CalledProcessError:
+        raise RuntimeError("Failed to install Ollama on Windows")
+    finally:
+        installer_path.unlink(missing_ok=True)
+
+def install_ollama_linux():
+    """Install Ollama on Linux using the official script."""
+    try:
+        subprocess.run("curl -fsSL https://ollama.com/install.sh | sh", shell=True, check=True)
```

Ruff (0.12.2)

`subprocess` call with `shell=True` seems safe, but may be changed in the future; consider rewriting without `shell`

(S602)